### DISCUSSION TITLE
Split long messages into multiple messages

### DISCUSSION THREAD

**Discussion by ghost** [nutgram/nutgram#403] [Ideas]
```php
$bot->sendMessage(str_repeat("a", 5000));
```
There is a maximum message length of 4096 characters.
This code will fail: `Bad Request: message is too long`
Add an option to automatically split messages into multiple messages and then send instead of giving an error.

Something like this:
![image](https://user-images.githubusercontent.com/78247242/225552441-0a71c609-3043-475d-acaa-cac12b3fff0e.png)

//------------------------------------------------------

```php
$tg->sendPhoto('photo', [
    'caption' => str_repeat("a", 5000),
]);
```
There is a maximum message length of 1024 characters `for regular users`.
For photos or other type of files, it is better to send the file with 1024 characters first and send the rest of the caption in response to the file.

Something like this:
![image](https://user-images.githubusercontent.com/78247242/225552030-cbba578c-58ab-4ea5-abcc-cd1b55065c6d.png)

**Comment by Lukasss93** [comment]
Already exist for sendMessage method: https://nutgram.dev/docs/configuration/installation#split_long_messages

**Reply by ghost** [reply]
Oh God !
Why is Nutgram so good? I can't believe  : )
I have seen and read the document many times... but every time new things appear :/

In any case, if it is only for sendMessage, it is good to add it for files as well

Thank you again .

**Reply by fossildj** [reply]
> Already exist for sendMessage method: https://nutgram.dev/docs/configuration/installation#split_long_messages

how do i use it? i tried and it doesnt work.

```
Next SergiX44\Nutgram\Telegram\Exceptions\TelegramException: Bad Request: message is too long in /home/mcsgarag/public_html/ben/bots/panoramabot/vendor/nutgram/nutgram/src/Telegram/Client.php:327
Stack trace:
```

`    $bot->sendMessage($result, ['parse_mode' => 'html','split_long_messages'=> true]);`

**Reply by Lukasss93** [reply]
Read the documentation!

```php
$bot = new Nutgram('you telegram token here', [
    'split_long_messages'=> true
]);
```

**Reply by fossildj** [reply]
> Read the documentation!
> 
> ```
> $bot = new Nutgram('you telegram token here', [
>     'split_long_messages'=> true
> ]);
> ```

sorry, i read it and though it apply to sendMessage parameters. my bad!
